lib/src/http/common/client.dart

```diff
@@ -93,7 +104,11 @@ abstract class CommonClient implements Client {
     if (withCredentials == true) {
       request.withCredentials = true;
     }
-    _requests.add(request);
+    request.autoRetry
+      ..enabled = autoRetry.enabled
+      ..forHttpMethods = autoRetry.forHttpMethods
+      ..forStatusCodes = autoRetry.forStatusCodes
+      ..test = autoRetry.test;
```

Comment:
Would it make sense to put this logic into an `AutoRetryConfig.clone()` method? Then you could do 

```dart
request.autoRetry = autoRetry.clone;
```

And avoid doing this in more than one place. Would make adding fields easier too, if that ever were to happen.

Just something to consider, this is fine too.